dev/dev-env-setup.sh

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+
+set +exv
+
+showHelp() {
+cat << EOF
+Usage: ./install-dev-env.sh [-d] [directory to install virtual environment] [-v] [-q]
+Development environment setup script for Python.
+This script will:
+
+  - Install pyenv if not installed
+  - Retrieve the appropriate Python version (minimum required) for compatibility support
+  - Check if the virtual environment already exists
+    - If it does, prompt for replacement
+      - If replacing, delete old virtual environment.
+  - Create a virtual environment using the minimum required Python version based on previous step logic
+  - Activate the environment
+  - Install required dependencies for the dev envrionment
+
+  Example usage:
+
+  From root of MLflow repository on local with a destination virtualenv path of <root>/.venvs/mlflow-dev:
+
+  dev/dev-env-setup.sh -d ~/.venvs/mlflow-dev
```

Comment:
I think the environment location should be `{path_to_mlflow_repository}/.venv` to make it easier to remove it.

Reply:
good idea! 
changed to 
`dev/dev-env-setup.sh -d $(pwd)/.venvs/mlflow-dev`